### ISSUE TITLE
stripchat.py

### ISSUE THREAD

**Issue by SnowManWithCarrot** [Damianonymous/streamlink-plugins#19]
The plugin will not work if the streamer name contains "-"

Example:

"My-name-is"

or

"My-name"

Error codes:

"error: Unable to validate JSON: Unable to validate key 'cam': Unable to validate result: Type of [] should be 'dict' but is 'list'"

**Comment by nxkbd** [comment]
@SnowManWithCarrot 
Workflow that works for me:
get latest `stripchat.py` and change line 7
`_url_re = re.compile(r"https?://(\w+\.)?stripchat\.com/(?P<username>\w+)")`
to
`_url_re = re.compile(r"https?://(\w+\.)?stripchat\.com/(?P<username>[\w\d_-]+)")`

**Comment by SnowManWithCarrot** [comment]
@nxkbd Thanks that actually fixed the problem :)

**Comment by mikedouglax** [comment]
@nxkbd  the same error happens when the streamer name contains "_"

**Comment by SnowManWithCarrot** [comment]
> 
> 
> @nxkbd the same error happens when the streamer name contains "_"

Apply the fix @nxkbd provided and it will also fix the "_" issue.

I've just tested both.

**Comment by mikedouglax** [comment]
i've just tried, let me try again

**Comment by mikedouglax** [comment]
error: Unable to open URL: https://b-hls-20.strpst.com/hls/29559103/master_29560103.m3u8 (HTTPSConnectionPool(host='b-hls-20.strpst.com', port=443): Read timed out. (read timeout=20.0))

**Comment by Damianonymous** [comment]
> error: Unable to open URL: https://b-hls-20.strpst.com/hls/29559103/master_29560103.m3u8 (HTTPSConnectionPool(host='b-hls-20.strpst.com', port=443): Read timed out. (read timeout=20.0))

**Comment by Damianonymous** [comment]
@mikedouglax

This is probably a different problem and also occurs with standard streamer names.

**Comment by SnowManWithCarrot** [comment]
@Damianonymous correct.

I can confirm that is happening with some of the streamers that mostly stream from outside and their connection is not that good so the calls from the API take longer than 20s leading to the "timeout 20.0" .

Possible solution is to remove the timeout or add sleep.time(xx).

**Comment by odnar-dev** [comment]
need to add default stream as a fallback .
i think i fixed it with [this](https://github.com/streamlink-plugins/streamlink-plugins/commit/ec3659b14033b81a745037788eadd5cac7a7b40c#diff-35f98d50f47358cb730ae6dd8954ee7d1d5a3e9376f5bf341fd9c747514886e2)

**Comment by SnowManWithCarrot** [comment]
I tested it for few days and can confirm that the fix @odnar-dev fixed the timeout issue.
